app.js

```diff
@@ -202,6 +205,54 @@ module.exports = ({ sharedPath: sharedPathIn, port, maxUploadSize, zipCompressio
     });
   }));
 
+
+  // zip file name function 
+  const zipName = () => new Date(Date.now() + 19800000).toISOString().slice(0, -5) + ".zip"
+
+  app.post('/api/zipfilesdownload', asyncHandler(async (req, res) => {
+    try {
+      let zipFileName = zipName();
+
+      const form = formidable({ multiples: true });
+      form.parse(req, (err, fields, filesList) => {
+        let obj = JSON.parse(fields.obj);
+
+        let files = obj.filesName;
+        let currentdir = obj.currentDir
+
+        res.setHeader('Content-Type', 'application/octet-stream');
+        res.setHeader('Content-Disposition', `attachment; filename="${zipFileName}"`);
+
+        const archive = archiver('zip', {
+          zlib: { level: 9 } // Sets the compression level
+        });
+
+        archive.on('error', (err) => {
+          console.error(err);
+          res.status(500).send({ error: 'Error creating the zip file' });
+        });
+
+        for (let i = 0; i < files.length; i++) {
+          const targetPath = join(__dirname, `.${currentdir}`, files[i]);
+          // const targetPath =  path.join(__dirname, `../tmp/resource/`, files[i]);
+          if (fs.existsSync(targetPath)) {
+            let realname = files[i];
+            archive.file(targetPath, { name: realname }); // Add each file to the archive
+          }
+        }
+        archive.pipe(res);
+        archive.finalize();
+      });
+
+    } catch (error) {
+      console.error("Error:", error);
+      res.status(500).send({ error: 'Internal server error' });
+    }
```

Comment:
this shouldn't be necessary because we already have an asyncHandler

Reply:
okay, I will remove that